controller/src/beerocks/master/db/db.cpp

```diff
@@ -2117,6 +2117,324 @@ bool db::get_hostap_is_acs_enabled(std::string mac)
+
+    auto scan_status =
+        (single_scan) ? n->hostap->single_scan_status : n->hostap->continuous_scan_status;
+    scan_status->last_scan_error_code = error_code;
+
+    return true;
+}
+
+beerocks::eChannelScanErrCode db::get_channel_scan_results_status(const std::string &mac,
+                                                                  const bool single_scan)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return beerocks::CHANNEL_SCAN_INTERNAL_FAILURE;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return beerocks::CHANNEL_SCAN_INTERNAL_FAILURE;
+    }
+
+    auto scan_status =
+        (single_scan) ? n->hostap->single_scan_status : n->hostap->continuous_scan_status;
+    return scan_status->last_scan_error_code;
+}
+
+bool db::set_channel_scan_dwell_time_msec(const std::string &mac, int dwell_time_msec,
+                                          bool single_scan)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+    LOG(DEBUG) << __FUNCTION__ << "," << (single_scan ? "single" : "continuous")
+               << ", dwell time msec = " << dwell_time_msec;
+
+    if (dwell_time_msec <= 0) {
+        LOG(ERROR) << __FUNCTION__ << "Only >0 dwell time is supported!";
+        return false;
+    }
+
+    auto scan_config =
+        (single_scan) ? n->hostap->single_scan_config : n->hostap->continuous_scan_config;
+    scan_config->dwell_time_msec = dwell_time_msec;
+
+    return true;
+}
+
+int db::get_channel_scan_dwell_time_msec(const std::string &mac, const bool single_scan)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return -1;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return -1;
+    }
+
+    auto scan_config =
+        (single_scan) ? n->hostap->single_scan_config : n->hostap->continuous_scan_config;
+    return scan_config->dwell_time_msec;
+}
+
+bool db::set_channel_scan_pool(const std::string &mac, const std::set<uint8_t> &channel_pool,
+                               const bool single_scan)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+
+    //Validate new dcs channel pool with hostap_supported_channels list
+    //This is a workaround
```

Comment:
This will be fixed as a separate task (already have one in the intel backlog)
start using GET_HW_FEATURES for channel scan pool validation.
I'll open task in github.

Reply:
opened #686